repos/system_upgrade/el7toel8/libraries/rhsm.py

```diff
@@ -291,37 +346,25 @@ def switched_certificate(context, rhsm_info, cert_path, version):
         if os.path.isdir(context.full_path(path)):
             context.copy_to(cert_path, os.path.join(path, os.path.basename(cert_path)))
 
-    unset_release(context)
-    try:
-        refresh(context)
-        # only ga has releases in rhsm
-        if get_product_type('target') == 'ga':
-            set_release(context, version)
-        target_rhsm_info = TargetRHSMInfo()
-        scan_rhsm_info(context, target_rhsm_info)
-        yield target_rhsm_info
-    finally:
-        # Restore backup of product certificates
-        context.call(['rm', '-rf', pki_path], checked=False)
-        context.call(['cp', '-a', pki_backup_path, pki_path], checked=False)
-        unset_release(context)
-        # Restore release - only ga has releases in rhsm
-        if get_product_type('source') == 'ga':
-            restore_release(context, rhsm_info)
+    # Restore product certificates from the backup
+    context.call(['rm', '-rf', pki_path], checked=False)
+    context.call(['cp', '-a', pki_backup_path, pki_path], checked=False)
```

Comment:
with this change, it's not context manager anymore. You will switch the ertificates, restore the original and then continue. contextmanager functions do not make sense without yield